### PR TITLE
Refs [TASK][34582] Show list recipe using Ingredients

### PR DESCRIPTION
**Pull request by phong10101998**
## Related Tickets
- [#Task 34582: Show list recipe using that Ingredients](https://edu-redmine.sun-asterisk.vn/issues/34582)
## WHAT
- Get Data recipe and bind data to view
- Load more
## Evidence (Screenshot or Video)
image
Cái Load more nhanh quá em chụp không có kịp :d :d 
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/spoonacular/ui/ingredient/adapter/RecipeByIngredientAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    class RecipeByIngredientViewHolder(
+        itemView: View, private val onItemClicked: (Recipe) -> Unit
+    ) : RecyclerView.ViewHolder(itemView) {
+
+        fun bindData(recipe: Recipe) {
+            itemView.apply {
+                Glide.with(context)
+                    .load(recipe.image)
+                    .into(imageSimilar)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/ingredient/adapter/RecipeByIngredientAdapter.kt]: viết ext đi

File: app/src/main/java/com/sun/spoonacular/utils/SafeLet.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.sun.spoonacular.utils
+
+
+inline fun <T, E, R> safeLet(a: T?, b: E?, block: (T, E) -> R) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/utils/SafeLet.kt:3]: viết cái gì đây chú?

Comment (phong10101998) [reply app/src/main/java/com/sun/spoonacular/utils/SafeLet.kt:3]: Check 2 cái list thôi a (list đầy đủ với list load more), em viết ra luôn để bên search với favourite dùng lại

File: app/src/main/java/com/sun/spoonacular/ui/ingredient/IngredientDetailViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import com.sun.spoonacular.data.source.Repository
+import com.sun.spoonacular.ui.base.BaseViewModel
+import com.sun.spoonacular.utils.Resource
+import com.sun.spoonacular.utils.safeLet
+import kotlinx.coroutines.*
+
+class IngredientDetailViewModel : BaseViewModel() {
+
+    private val repository = Repository.getInstance()
+    val recipe = MutableLiveData<MutableList<Recipe?>>()
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/ingredient/IngredientDetailViewModel.kt]:
```suggestion
    val recipes = MutableLiveData<MutableList<Recipe?>>()
```

File: app/src/main/java/com/sun/spoonacular/ui/ingredient/IngredientDetailViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    val recipeAll: MutableLiveData<MutableList<Recipe?>>
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/ingredient/IngredientDetailViewModel.kt]:
```suggestion
    val recipesAll: MutableLiveData<MutableList<Recipe?>>
```

File: app/src/main/java/com/sun/spoonacular/ui/ingredient/IngredientDetailViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        get() = _recipeAll
+    private val _recipeAll = MutableLiveData<MutableList<Recipe?>>()
+    var checkLoadMore = MutableLiveData<Boolean>()
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/ingredient/IngredientDetailViewModel.kt]:
```suggestion
    var isLoadMore = MutableLiveData<Boolean>()
```

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**